libparsec/crates/client/src/workspace/merge.rs

```diff
@@ -150,7 +150,19 @@ pub(super) fn merge_local_folder_manifest(
     prevent_sync_pattern: Option<&Regex>,
     local: &LocalFolderManifest,
     remote: FolderManifest,
+    force_apply_pattern: bool,
 ) -> MergeLocalFolderManifestOutcome {
+    // 0) Sanity checks, caller is responsible to handle them properly !
+    debug_assert_eq!(local.base.id, remote.id);
+
+    let cleaned_up_local_manifest = if force_apply_pattern {
+        Some(local.apply_prevent_sync_pattern(prevent_sync_pattern, timestamp))
+    } else {
+        None
+    };
+
+    let local_manifest = cleaned_up_local_manifest.as_ref().unwrap_or(local);
```

Comment:
```suggestion
    let local = cleaned_up_local_manifest.as_ref().unwrap_or(local);
```
we have `remote`, `local` & `merge_in_progress` so I think it's more consistent to avoid using `_manifest` suffix here and just stick with `local`